### ISSUE TITLE
Paginate

### ISSUE THREAD

**Issue by dsbilling** [tzookb/tbmsg#22]
Is it possible to add a paginate function for messages in convos and for conversations?

**Comment by abhishek238** [comment]
it looks good, can you please suggest me, how can i work group messaging with this package

**Comment by tzookb** [comment]
@abhishek238 please open a new issue, so we could discuss there.

**Comment by RaoulDijksman** [comment]
This is something i ran into my self! I have been looking into it. Shouldn't be that hard to make but depends on how far you want to take it. Making a function like this should get people get people started:
```php
public static function getMessages($convId, $start = 0, $perPage = 20) {
   // SQL QUERY LIMIT = $perPage and OFFSET = $start.
}
```
This way you won't need to load all messages but the only problem with this is that you won't know how many pages there are in total. This also  means you won't be able to make a Laravel Paginator object. I don't like working with the Paginator for messages and don't recommend it. But if that is what you want you could make something like:
```php
$conv = TBMsg::getConversationMessages($convId, $userId,);
$conv->getAllMessages();
$conv->paginateMessages($perPage);

-----------------------

public function paginateMessages($perPage = 20) {
  // LIMIT = $perPage and OFFSET = $start.
  // $messages= all the messages
  // $total = count() on all the $messages
  // Paginator::make($collection->all(), $total, $perPage);
}
```
The problem here is that Pagenator::make() will use Paginator::getCurrentPage() to get the current page from the GET parameters. This will also make the package more dependent.

Raoul

**Comment by tzookb** [comment]
yes your first option will be the option, as I dont want to add the laravel paginator now.

Ill implement it soon

**Comment by tzookb** [comment]
just finished the big change with tests, so now it will be much safer to change stuff.
I'll try to add it soon

**Comment by tmunzar-jfish** [comment]
Yes, this feature is very much required to move forward.
I'd love to help out implementing this if you don't have the time.

**Comment by tzookb** [comment]
Hi @tmunzar-jfish I would always would be happy  to get help :)

**Comment by rohitnaidu19** [comment]
Is this feature implemented yet?

**Comment by tzookb** [comment]
not yet, hope to do it soon, or would be happy if you have suggestions.

**Comment by rohitnaidu19** [comment]
Your package is awesome. One of the best chat systems for laravel
available. Very flexible. Just a few issues else it's perfect.

Wish I could help but I'm a noob :-)
